Sources/FunctionalTableData/TableView/FunctionalTableData+UITableViewDataSource.swift

```diff
@@ -57,4 +57,347 @@ extension FunctionalTableData {
+			exceptionHandler.handle(exception: exception)
+		}
+	}
+	
+	@available(iOS 13.0, *)
+	class DiffableDataSource: UITableViewDiffableDataSource<DiffableTableSection, AnyCellConfigType> {
+		let cellStyler: CellStyler
+		
+		var data: TableData {
+			return cellStyler.data
+		}
+		
+		init(tableView: UITableView, cellStyler: CellStyler) {
+			self.cellStyler = cellStyler
+			super.init(tableView: tableView) { (tableView, indexPath, cellConfigType) in
+				let sectionData = cellStyler.data.sections[indexPath.section]
+				let cell = cellConfigType.dequeueCell(from: tableView, at: indexPath)
+				  let accessibilityIdentifier = ItemPath(sectionKey: sectionData.key, itemKey: cellConfigType.key).description
+				  cellConfigType.accessibility.with(defaultIdentifier: accessibilityIdentifier).apply(to: cell)
+				  cellStyler.update(cell: cell, cellConfig: cellConfigType, at: indexPath, in: tableView)
+				  return cell
+			}
+		}
+		
+		override func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
+			guard let cellConfig = data.sections[indexPath] else { return false }
+			return cellConfig.actions.hasEditActions || self.tableView(tableView, canMoveRowAt: indexPath) || cellConfig.style?.selected != nil
+		}
+		
+		override func tableView(_ tableView: UITableView, canMoveRowAt indexPath: IndexPath) -> Bool {
+			return data.sections[indexPath]?.actions.canBeMoved ?? false
+		}
+		
+		override func tableView(_ tableView: UITableView, moveRowAt sourceIndexPath: IndexPath, to destinationIndexPath: IndexPath) {
+			// Should only ever be moving within section
+			assert(sourceIndexPath.section == destinationIndexPath.section)
+			
+			// Update internal state to match move
+			let cell = cellStyler.data.sections[sourceIndexPath.section].rows.remove(at: sourceIndexPath.row)
+			cellStyler.data.sections[destinationIndexPath.section].rows.insert(cell, at: destinationIndexPath.row)
+			cellStyler.data.sections[sourceIndexPath.section].didMoveRow?(sourceIndexPath.row, destinationIndexPath.row)
+			super.tableView(tableView, moveRowAt: sourceIndexPath, to: destinationIndexPath)
+		}
+	}
+	
+	@available(iOS 13.0, *)
+	class DiffableDataSourceFunctionalTableDataImpl: FunctionalTableDataImpl {
+		var datasource: DiffableDataSource!
+		var isRendering: Bool = false
+		let name: String
+		let cellStyler: CellStyler
+		let applyQueue: DispatchQueue
+		
+		var tableView: UITableView? {
+			didSet {
+				guard let tableView = tableView else { return }
+				let dataSource = DiffableDataSource(tableView: tableView, cellStyler: cellStyler)
+				dataSource.defaultRowAnimation = .none
+				self.datasource = dataSource
+				
+			}
+		}
+		
+		init(name: String, cellStyler: CellStyler) {
+			self.name = name
+			self.cellStyler = cellStyler
+			self.applyQueue = DispatchQueue.global(qos: .background)
```

Comment:
I imagine that snapshot applying is frequently tied to the user's interactions. Is there another queue with higher priority we could use? Maybe `.userInteractive`?

Reply:
Deleted this. Wasn't using the queue because it broke tests and didn't notice improvements.